docs/src/man/comparisons.md

```diff
@@ -235,6 +235,66 @@ The table below compares more advanced commands:
+| Reduce multiple values      | `df[, list(mean(x)), by = list(id) ]`             | `combine(groupby(df, :id), :x => mean)`   |
+| Add new columns (in place)  | `df[, x_mean := mean(x), by = list(id) ]`         | `transform!(groupby(df, :id), :x => mean)`|
+| Pick & transform columns    | `df[, list(x_mean = mean(x), y), by = list(id) ]` | `select(groupby(df, :id), :x => mean, :y)`|
+
+### More advanced commands
+
+| Operation                         | data.table                                                                                 | DataFrames.jl                                                               |
+|:----------------------------------|:-------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------|
+| Complex Function                  | `df[, list(mean(x, na.rm = T)) ]`                                                          | `combine(df, :x => x -> mean(skipmissing(x)))`                              |
+| Transform certain rows (in place) | `df[x<=0, x:=0 ]`                                                                          | `df.x[df.x .<= 0] .= 0`                                                     |
+| Transform several columns         | `df[, list(max(x), min(y)) ]`                                                              | `combine(df, :x => maximum, :y => minimum)`                                 |
+|                                   | `df[, lapply(.SD, mean), .SDcols = c("x", "y") ]`                                          | `combine(df, [:x, :y] .=> mean)`                                            |
+|                                   | `df[, lapply(.SD, mean), .SDcols = patterns("x*") ]`                                       | `combine(df, names(df, r"^x") .=> mean)`                                    |
+|                                   | `df[, unlist(lapply(.SD, function(x) c(max=max(x), min=min(x)))), .SDcols = c("x", "y") ]` | `combine(df, ([:x, :y] .=> [maximum minimum])...)`                          |
+| Multivariate function             | `df[, list(cor(x,y)) ]`                                                                    | `transform(df, [:x, :y] => cor)`                                            |
+| Row-wise                          | `df[, min_xy := min(x, y), by = 1:nrow(df)]`                                               | `transform(df, [:x, :y] => ByRow(min))`                                     |
+|                                   | `df[, argmax_xy := which.max(.SD) , .SDcols = patterns("x*"), by = 1:nrow(df) ]`           | `transform(df, AsTable(r"^x") => ByRow(argmax))`                            |
+| DataFrame as output               | `df[, .SD[, list(value = c(min(x), max(x)) )] ]`                                           | `combine(df, :x => (x -> (value = [minimum(x), maximum(x)],)) => AsTable)`  |
```

Comment:
`df[, .(c(min(x), max(x)))]`

Reply:
I was trying to show a typical use case for .SD
I typically find it useful for non-standard function that I have to execute by groups (a regression for example). But that's the best example I could come up with. 
If you have an idea, let me know; if not I will likely remove this row because this has generated some confusion on the goal before.

Reply:
Ok I see. One use-case for `.SD` that springs to mind is `DT[, .SD[1], by=group]`;  i.e. return the first row of every group.  Or, `DT[, .SD[which.max(someCol)], by=group]`; i.e. return the row in each group that has the biggest value in some column.

Reply:
Thanks. I think I confused myself with the dataframes syntax. Here are your examples, it would be great if some of the dataframes could tell me if this is idiomatic

First element by group (usually comes after a sort)
```R
df[, .SD[1], by=grp]
```
```Julia
combine(groupby(df, :grp), names(df) .=> (x->x[1]) .=> names(df))
```

Or max by group:
```R
df[, .SD[which.max(x)], by=grp]
```
```Julia
subset(groupby(df, :grp), :x => (x->(x.==maximum(x))) )
```

I am not sure the last one is the best way of doing this. But that is also a use case for subset by group (which @matthieugomez raised [here](https://github.com/JuliaData/DataFrames.jl/issues/2740) recently)

Reply:
> `combine(groupby(df, :grp), names(df) .=> (x->x[1]) .=> names(df))`

is just `combine(groupby(df, :grp), first)`

> `subset(groupby(df, :grp), :x => (x->(x.==maximum(x))) )`
is probably more natural to write as `combine(groupby(df, :grp), sdf -> sdf[argmax(sdf.x), :])`

(note that in both cases the approach in data.table and in DataFrames.jl is conceptually the same (I have learned something 😄) only the syntax is a bit different)

Reply:
That's amazing. I had no idea about `sdf`. Is this a reserved variable that stands for the df for each group within the scope of combine? I did not see it documented. 
That's exactly the kind of things I wanted to show. This will make the guide extra useful!

Reply:
No, the syntax `x -> f(x)` is used for creating an anonymous function. So `sdf` is argument to a lamdba.

Reply:
Just to expand on what @pdeffebach commented. We often use `sdf` to signal that this is a view of an original data frame, which is of `SubDataFrame` type.

Reply:
Thanks. I got totally confused by the lambda argument name and thought somehow this was a special variable much like `.SD` 
I am still amazed that this ends up being so easy. Kudos to DataFrames!